datadog_checks_base/datadog_checks/base/utils/db/query.py

```diff
@@ -34,6 +34,11 @@ def __init__(self, query_data):
         self.extra_transformers = None  # type: List[Tuple[str, Transformer]]
         # Contains the tags defined in query_data, more tags can be added later from the query result
         self.base_tags = None  # type: List[str]
+        # The collecton interval (in seconds) of the query. If None, the query will be run every check run.
+        self.collection_interval = None  # type: int
+        # The last time the query was executed. If None, the query has never been executed.
+        # This is only used when the collection_interval is not None.
+        self.last_execution_time = None  # type: float
```

Comment:
nit: should probably be a private attribute

Reply:
good suggestion. I wrap the check inside a `should_execute` function so that the execution timestamp is private to the query object. 
https://github.com/DataDog/integrations-core/pull/16765/files#diff-176155c9333df9505a51fb382db122c3b88b75a865c2614a24d0a07909e6096fR223-R239